code/game/objects/structures/curtains.dm

```diff
@@ -1,3 +1,71 @@
+//
+// Curtain types declaration
+//
+/decl/curtain_kind
+	var/name = "curtain"
+	var/color = "white"
+	var/alpha = 255
+	var/material_key = /decl/material/solid/plastic
+
+/decl/curtain_kind/proc/make_item(var/loc)
+	var/obj/item/curtain/C = new(loc, material_key, src)
+	C.alpha = alpha
+	C.set_color(color)
+	C.SetName("rolled [name]")
+	return C
+
+/decl/curtain_kind/proc/make_structure(var/loc, var/dir, var/opened = FALSE)
+	var/obj/structure/curtain/C = new(loc, dir, material_key, null, src)
+	C.alpha = alpha
+	C.set_color(color)
+	C.SetName(name)
+	C.set_opacity(opened)
+	return C
+
+//Cloth curtains
+/decl/curtain_kind/cloth
+	material_key = /decl/material/solid/cloth
+
+/decl/curtain_kind/cloth/bed
+	name = "bed curtain"
+	color = "#854636"
+
+/decl/curtain_kind/cloth/black
+	name = "black curtain"
+	color = "#222222"
+
+/decl/curtain_kind/cloth/bar
+	name = "bar curtain"
+	color = "#854636"
+
+//Plastic curtains
+/decl/curtain_kind/plastic
+	name = "plastic curtain"
+	color = "#b8f5e3"
+	material_key = /decl/material/solid/plastic
+
+/decl/curtain_kind/plastic/medical
+	alpha = 200
+
+/decl/curtain_kind/plastic/privacy
+	name = "privacy curtain"
+
+/decl/curtain_kind/plastic/shower
+	name = "shower curtain"
+	color = "#acd1e9"
+	alpha = 200
+
+/decl/curtain_kind/plastic/shower/engineering
+	color = "#ffa500"
+
+/decl/curtain_kind/plastic/shower/security
+	color = "#aa0000"
+
+/decl/curtain_kind/plastic/canteen
+	name = "privacy curtain"
+	color = COLOR_BLUE_GRAY
+
+
 /obj/item/curtain
```

Comment:
While you're in the area it might be worth doing something like
````
/obj/item/curtain/Initialize()
  . = ..()
  var/decl/curtain_kind/curtain_decl = GET_DECL(curtain_kind)
  matter = atom_info_repository.get_matter_for(/obj/structure/curtain, curtain_decl.material_key)
````
Otherwise the recipe cost population might not function correctly.

Reply:
Is that recipe cost population thing new?

Reply:
It's Nebula specific but it's something like a year old.

Reply:
Turns out these items aren't used by the recipes and I'm dumb.